src/web.ts

```diff
@@ -56,43 +56,28 @@ export class CapacitorFirestoreWeb
     });
   }
 
-  public initializeFirestore(options: FirestoreConfig): Promise<void> {
-    let teardownPromise = Promise.resolve();
-
+  public async initializeFirestore(options: FirestoreConfig): Promise<void> {
```

Comment:
Should the return type be changed when using `async`? The return type of the method would not just be `void`? but actually transpiles as `Promise<void>`.

Also I seem to remember some issue that occurred when we used `async` instead of chaining but happy to give it a go

Reply:
No need to change the return type. The return type for `async` functions is a `Promise`.
It's a common practice to use `async`/`await` rather than `.then`.
I have tested the changes multiple times for multiple cases. But a confirmation from your end is needed.